falcon/redirects.py

```diff
@@ -0,0 +1,148 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import falcon
+from falcon.http_status import HTTPStatus
+
+
+class HTTPMovedPermanently(HTTPStatus):
+    """301 Moved Permanently.
+
+    The 301 (Moved Permanently) status code indicates that the target
+    resource has been assigned a new permanent URI.
+
+    Note:
+        For historical reasons, a user agent MAY change the request
+        method from POST to GET for the subsequent request.  If this
+        behavior is undesired, the 308 (Permanent Redirect) status code
+        can be used instead.
+
+    See also: https://tools.ietf.org/html/rfc7231#section-6.4.2
+
+    Args:
+        location (str): URI to provide as the Location header in the
+            response.
+    """
+
+    def __init__(self, location):
+        super(HTTPMovedPermanently, self).__init__(
+            falcon.HTTP_301, {'location': location})
+
+
+class HTTPFound(HTTPStatus):
+    """302 Found.
+
+    The 302 (Found) status code indicates that the target resource
+    resides temporarily under a different URI.  Since the redirection
+    might be altered on occasion, the client ought to continue to use the
+    effective request URI for future requests.
+
+    Note:
+        For historical reasons, a user agent MAY change the request
+        method from POST to GET for the subsequent request.  If this
+        behavior is undesired, the 307 (Temporary Redirect) status code
+        can be used instead.
+
+    See also: https://tools.ietf.org/html/rfc7231#section-6.4.3
+
+    Args:
+        location (str): URI to provide as the Location header in the
+            response.
+    """
+
+    def __init__(self, location):
+        super(HTTPFound, self).__init__(
+            falcon.HTTP_302, {'location': location})
+
+
+class HTTPSeeOther(HTTPStatus):
+    """303 See Other.
+
+    The 303 (See Other) status code indicates that the server is
+    redirecting the user agent to a *different* resource, as indicated
+    by a URI in the Location header field, which is intended to provide
+    an indirect response to the original request.
+
+    A 303 response to a GET request indicates that the origin server
+    does not have a representation of the target resource that can be
+    transferred by the server over HTTP.  However, the Location field
+    value refers to a resource that is descriptive of the target
+    resource, such that making a retrieval request on that other
+    resource might result in a representation that is useful to
+    recipients without implying that it represents the original target
+    resource.
```

Comment:
That is one long sentence. I had to re-read it a couple times to fully understand the point. If people have a problem understanding this, you might consider rephrasing or breaking this sentence up.